### PR TITLE
When initializing a task, write node version to airplane.config.yaml

### PR DESCRIPTION
**Pull request by leeweisberger**
## Description
When creating a new task, create a new airplane.config.yaml and write the node version there
When there is already an airplane.config.yaml with no node version, update that airplane.config.yaml with the correct node version
When there is already an airplane.config.yaml or package.json with a conflicting node version, do nothing and warn

## Test plan
Manually tested all of the above cases

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -17,53 +17,126 @@ import (
...
+type PackageJSONOptions = struct {
+	Dependencies NodeDependencies
+}
+
 // CreatePackageJSON ensures there is a package.json in path with the provided dependencies installed.
 // If package.json exists in cwd, use it.
 // If package.json exists in parent directory, ask user if they want to use that or create a new one.
 // If package.json doesn't exist, create a new one.
-func CreatePackageJSON(directory string, dependencies NodeDependencies) error {
+func CreatePackageJSON(directory string, packageJSONOptions PackageJSONOptions) error {
```
Comment (leeweisberger) [inline pkg/node/node.go:40]: This function is just a refactor. No functionality change.

File: cmd/airplane/tasks/initcmd/init.go [hunk 2]
```diff
@@ -836,21 +836,37 @@ func apiTaskToRuntimeTask(task *libapi.Task) *runtime.Task {
...
+			},
+		}); err != nil {
+			return err
+		}
+
+		_, nodeVersion, err := def.GetBuildType()
+		if err != nil {
+			return err
+		}
+		if err := node.CreateAirplaneConfig(cwd, isNew, deployconfig.AirplaneConfig{
```
Comment (ameyakhare) [inline cmd/airplane/tasks/initcmd/init.go:860]: what does isNew do here?

Comment (ameyakhare) [inline cmd/airplane/tasks/initcmd/init.go:860]: shouldnt we not be creating this if it already exists?

File: pkg/node/node.go [hunk 3]
```diff
@@ -263,27 +336,27 @@ func mergeTSConfig(configFile []byte, strategy MergeStrategy) error {
...
 		}
-		logger.Step("Added tsconfig.json...")
+		logger.Step("Created tsconfig.json")
 	}
 
 	return nil
 }
 
-func mergeTSConfigsRecursively(dest, src map[string]interface{}) {
+func mergeStructsRecursively(dest, src map[string]interface{}) {
```
Comment (ameyakhare) [inline pkg/node/node.go:351]: nit: these arent really structs so if you want to generalize, make it mergeMapsRecusively

File: cmd/airplane/tasks/initcmd/init.go (hunk 2 above)
Comment (ameyakhare) [reply cmd/airplane/tasks/initcmd/init.go:860]: nvm, it looks like CreateAirplaneConfig checks if there is already another one, a little confusing though imo based on the function name (maybe CreateOrMergeAirplaneConfig? idk)

**Review by ameyakhare — CHANGES_REQUESTED** [review]
some open questions

File: cmd/airplane/tasks/initcmd/init.go (hunk 2 above)
Comment (leeweisberger) [reply cmd/airplane/tasks/initcmd/init.go]: I updated the code to remove it. Don't think it's necessary

Comment (leeweisberger) [reply cmd/airplane/tasks/initcmd/init.go]: Yeah updated

**Review by ameyakhare — APPROVED** [review]
nice

**Pull request merged**